modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/utils/ServerUtilities.java

```diff
@@ -0,0 +1,364 @@
+		HttpGet httpGet = new HttpGet(url);
+		HttpGet httpGetWithHeaders = (HttpGet) buildHeaders(httpGet, headers, HTTP_METHODS.GET);
+		HttpClient httpClient = getCertifiedHttpClient();
+		Map<String, String> responseParams = new HashMap<String, String>();
+
+		try {
+			HttpResponse response = httpClient.execute(httpGetWithHeaders);
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
+			responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+			                   String.valueOf(response.getStatusLine().getStatusCode()));
+		} catch (ClientProtocolException e) {
+			throw new IDPTokenManagerException("Invalid client protocol.", e);
+		} catch (IOException e) {
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
+			responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+			                   Constants.INTERNAL_SERVER_ERROR);
+			throw new IDPTokenManagerException("Server connectivity failure.", e);
+		}
+		
+		return responseParams;
+	}
+	
+	public static Map<String, String> sendPostRequest(String url, JSONObject params, Map<String, String> headers) throws IDPTokenManagerException{
+		HttpPost httpPost = new HttpPost(url);
+		HttpClient httpClient = getCertifiedHttpClient();
+		Map<String, String> responseParams = new HashMap<String, String>();
+		
+		if (params != null) {
+			try {
+				httpPost.setEntity(new StringEntity(params.toString()));
+			} catch (UnsupportedEncodingException e) {
+				throw new IDPTokenManagerException("Invalid encoding type.", e);
+			}
+		}
+
+		HttpPost httpPostWithHeaders = (HttpPost) buildHeaders(httpPost, headers, HTTP_METHODS.POST);
+		try {
+			HttpResponse response = httpClient.execute(httpPostWithHeaders);
+			String status = String.valueOf(response.getStatusLine().getStatusCode());
+
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
+			responseParams.put(Constants.SERVER_RESPONSE_STATUS, status);
+
+		} catch (ClientProtocolException e) {
+			throw new IDPTokenManagerException("Invalid client protocol.", e);
+		} catch (IOException e) {
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
+			responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+			                   Constants.INTERNAL_SERVER_ERROR);
+			throw new IDPTokenManagerException("Server connectivity failure.", e);
+		}
+		
+		return responseParams;
+	}
+
+	public static Map<String, String> postDataAPI(EndPointInfo apiUtilities,
+	                                              Map<String, String> headers)
+			throws IDPTokenManagerException {
+		HTTP_METHODS httpMethod = apiUtilities.getHttpMethod();
+		String url = apiUtilities.getEndPoint();
+		Map<String, String> params = apiUtilities.getRequestParamsMap();
+
+		Map<String, String> responseParams = new HashMap<String, String>();
+		HttpClient httpclient = getCertifiedHttpClient();
+		String payload = buildPayload(params);
+		
+		if (httpMethod.equals(HTTP_METHODS.POST)) {		
```

Comment:
As a best practice, whenever possible compare the value assigned to a variable against a known constant.